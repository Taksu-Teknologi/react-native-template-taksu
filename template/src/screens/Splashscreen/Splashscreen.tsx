import { StackNavigationProp } from '@react-navigation/stack';
import React, { useEffect } from 'react';
import { StatusBar, StyleSheet, View } from 'react-native';
import * as Animatable from 'react-native-animatable';
import color from '../../constants/color';
import { sh, sw } from '../../utils/dimensions';
import { RootStackParamList } from '../../utils/Navigation';

interface Props {
    navigation: StackNavigationProp<RootStackParamList, 'Splashscreen'>;
}

const Splashscreen = (props: Props) => {
    useEffect(() => {
        setTimeout(() => {
            props.navigation.replace('Dashboard');
        }, 3000);
    }, []);

    return (
        <>
            <StatusBar barStyle="dark-content" />
            <View style={styles.container}>
                <Animatable.Image
                    useNativeDriver
                    animation="fadeInUp"
                    duration={400}
                    delay={200}
                    source={require('../../images/logo.png')}
                    resizeMode="contain"
                />
            </View>
        </>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: color.PRIMARY,
    },
    logo: {
        width: sw(400),
        height: sh(200),
    },
});

export default Splashscreen;
