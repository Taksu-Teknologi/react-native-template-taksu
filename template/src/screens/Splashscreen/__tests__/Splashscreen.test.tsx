import { shallow } from 'enzyme';
import React from 'react';
import Splashscreen from '../Splashscreen';

describe('Splash Screen', () => {
    it('renders correctly', () => {
        shallow(<Splashscreen navigation={{} as any} />);
    });

    it('take snapshot', () => {
        const wrapper = shallow(<Splashscreen navigation={{} as any} />);

        expect(wrapper).toMatchSnapshot();
    });
});
