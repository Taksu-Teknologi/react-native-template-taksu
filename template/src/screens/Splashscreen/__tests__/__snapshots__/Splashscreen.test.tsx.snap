// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen take snapshot 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    barStyle="dark-content"
    showHideTransition="fade"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <withAnimatable(Image)
      animation="fadeInUp"
      delay={200}
      direction="normal"
      duration={400}
      iterationCount={1}
      iterationDelay={0}
      onAnimationBegin={[Function]}
      onAnimationEnd={[Function]}
      onTransitionBegin={[Function]}
      onTransitionEnd={[Function]}
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/images/logo.png",
        }
      }
      useNativeDriver={true}
    />
  </View>
</Fragment>
`;
